classes/author/Repository.inc.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * @file classes/author/Repository.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class author
+ *
+ * @brief A repository to find and manage authors.
+ */
+
+namespace APP\author;
+
+use APP\core\Request;
+use PKP\Services\PKPSchemaService;
+use Illuminate\Support\Facades\App;
+
+class Repository extends \PKP\author\Repository
+{
+    public function __construct(DAO $dao, Request $request, PKPSchemaService $schemaService)
+    {
+        parent::__construct($dao, $request, $schemaService);
+    }
+
+    /** @copydoc DAO::getCollector() */
+    public function getCollector(): Collector
+    {
+        return App::make(Collector::class);
+    }
+
+    /**
+     * Add an author to a chapter.
+     */
+    public function addToChapter(int $authorId, int $chapterId, bool $isPrimary = false, int $sequence = 0)
+    {
+        $this->dao->insertChapterAuthor(
+            $authorId,
+            $chapterId,
+            $isPrimary,
+            $sequence
+        );
+    }
```

Comment:
I can see that this is used in the import/export toolset, where you only have an ID. So i think it's ok to leave alone. Generally, though, I prefer to write the methods to expect objects where possible. That way if we need to extend the method to write a log or fire a hook, we have the full objects available to us.